thread/timer.h

```diff
@@ -39,6 +39,7 @@ namespace photon
         {
             _on_timer = on_timer;
             _default_timeout = default_timeout;
+            _reset_timeout = -1UL;
```

Comment:
https://github.com/alibaba/PhotonLibOS/blob/7c597803832bef0a11b20ba9b8e5f2aa41b539a9/thread/thread.cpp#L1459

Reply:
In timer stub, retry use `_reset_timeout` only when timer function returns a negative result. User should not interrupt timer with EAGAIN in any cases, but only call `Timer::reset(new_timeout)` method.

The `_reset_timeout` only be set when `reset(new_timeout)` is called. Set in constructor is meaningless.